core/src/main/java/org/elasticsearch/index/IndexService.java

```diff
@@ -606,7 +608,11 @@ private Query parse(AliasMetaData alias, QueryShardContext shardContext) {
         try {
             byte[] filterSource = alias.filter().uncompressed();
             try (XContentParser parser = XContentFactory.xContent(filterSource).createParser(filterSource)) {
-                ParsedQuery parsedFilter = shardContext.toFilter(shardContext.newParseContext(parser).parseInnerQueryBuilder());
+                Optional<QueryBuilder> innerQueryBuilder = shardContext.newParseContext(parser).parseInnerQueryBuilder();
+                ParsedQuery parsedFilter = null;
+                if (innerQueryBuilder.isPresent()) {
+                    parsedFilter = shardContext.toFilter(innerQueryBuilder.get());
+                }
```

Comment:
I checked a few and found that e.g. `CommonTermsQueryBuilder#doToQuery()` can return null if the analyzer for the query doesn't provide any token. In `QueryShardContext#toQuery()` we do an explicit null-check and emit a MatchNoDocsQuery instead of null Query, in `toFilter()` we return null. I wonder if we should also return MatchNoDocsQuery there instead, then I could change the code above wo omitt the check, wdyt?